paddle/majel/malloc.cc

```diff
@@ -0,0 +1,131 @@
+#include "malloc.h"
+#include <glog/logging.h>
+
+#ifndef PADDLE_ONLY_CPU
+#include <cuda_runtime.h>
+#endif
+
+#define CHECK_CUDA(cudaFunc)                                         \
```

Comment:
In terms of C library design, let's follow C's convention of [malloc](http://www.cplusplus.com/reference/cstdlib/malloc/) -- returns NULL when the allocation fails.